src/sources/prometheus/parser.rs

```diff
@@ -1,301 +1,44 @@
-
-    let mut current_header = ParserHeader {
-        name: "".into(),
-        kind: ParserType::Untyped,
-    };
-
-    for line in packet.lines() {
-        let line = line.trim();
-        let line: String = WHITESPACE.replace_all(&line, " ").to_string();
-
-        if line.is_empty() {
-            continue;
-        }
-
-        if is_header(&line) {
-            // parse expected name and type from TYPE header
-            let header = parse_header(&line)?;
-            if !result.contains_key(&header) {
-                result.insert(header.clone(), Vec::new());
-            }
-            // we will need it to analyse the consequent lines
-            current_header = header;
-        } else if is_comment(&line) {
-            // skip comments and HELP strings
-        } else {
-            // parse the data line
-            let metric = parse_metric(&line)?;
-
-            current_header = match current_header.kind {
-                ParserType::Histogram => {
-                    // check if this is still a histogram
-                    if metric.name == format!("{}_bucket", current_header.name)
-                        || metric.name == format!("{}_count", current_header.name)
-                        || metric.name == format!("{}_sum", current_header.name)
-                    {
-                        current_header
-                    } else {
-                        // nope it's a new unrelated metric
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-                ParserType::Summary => {
-                    // check if this is still a summary
-                    if metric.name == current_header.name
-                        || metric.name == format!("{}_count", current_header.name)
-                        || metric.name == format!("{}_sum", current_header.name)
-                    {
-                        current_header
-                    } else {
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-                _ => {
-                    if metric.name == current_header.name {
-                        current_header
-                    } else {
-                        ParserHeader {
-                            name: metric.name,
-                            kind: ParserType::Untyped,
-                        }
-                    }
-                }
-            };
-
-            if let Some(lines) = result.get_mut(&current_header) {
-                lines.push(line);
-            } else {
-                result.insert(current_header.clone(), vec![line]);
-            }
-        }
-    }
-
-    Ok(result)
 }
 
 pub fn parse(packet: &str) -> Result<Vec<Metric>, ParserError> {
```

Comment:
The logic of this function is strongly tied into that of the rest of the parsing that was moved into a separate workspace. Was there a reason to keep this one here?

Reply:
Right, I've been working on separating out the whole event tree, but until that point this part has to stay here.